builtin/providers/aws/resource_aws_iam_group.go

```diff
@@ -93,6 +92,30 @@ func resourceAwsIamGroupReadResult(d *schema.ResourceData, group *iam.Group) err
 	return nil
 }
 
+func resourceAwsIamGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	if d.HasChange("name") || d.HasChange("path") {
+		iamconn := meta.(*AWSClient).iamconn
+		on, nn := d.GetChange("name")
+		op, np := d.GetChange("path")
+		fmt.Println(on, nn, op, np)
+		request := &iam.UpdateGroupInput{
+			GroupName:    aws.String(on.(string)),
+			NewGroupName: aws.String(nn.(string)),
+			NewPath:      aws.String(np.(string)),
+		}
+		_, err := iamconn.UpdateGroup(request)
+		if err != nil {
+			if iamerr, ok := err.(awserr.Error); ok && iamerr.Code() == "NoSuchEntity" {
+				d.SetId("")
+				return nil
```

Comment:
This would be a fine thing to do in the ``Read`` function, but I don't think this is the best behavior here: if (for some reason) the group gets deleted between the creation of the plan and the application of the plan then this will result in a successful exit but will leave the user with no group resource, even though the config says that the resource should exist. This is confusing, since users will expect the resource to match their config if ``terraform apply`` exits without error.

Instead I think it's better to actually fail in this case so we tell the user that it wasn't possible to converge on the settings in the config. If the user then makes a new plan the ``Read`` function will detect the resource is gone and the diff will include a create action as expected, allowing Terraform to create a fresh resource.